'SDL2/SDL_mixer.h' file not found

#9982
This fixes USE_SDL_MIXER = 2.

However, 'SDL2/SDL_mixer.h' file not found

<pre>
emcc -c main.c -o main.bc -O2 -s USE_SDL=2 -s USE_SDL_IMAGE=2 -s SDL2_IMAGE_FORMATS="['png']" -s USE_SDL_MIXER=2 -s USE_SDL_TTF=2
In file included from main.c:2:
./header.h:7:10: fatal error: 'SDL2/SDL_mixer.h' file not found
</pre>

[https://emscripten.org/docs/getting_started/FAQ.html](https://emscripten.org/docs/getting_started/FAQ.html)
I'm doing it as written here.

--- a/tests/sdl2_mixer.c
+++ b/tests/sdl2_mixer.c
@@ -15,7 +15,9 @@ void sound_loop_then_quit() {
 
     emscripten_cancel_main_loop();
     printf("Shutting down\n");
+#ifdef REPORT_RESULT
     REPORT_RESULT(1);
+#endif
 }
 
 int main(int argc, char* argv[]){

--- a/tests/test_other.py
+++ b/tests/test_other.py
@@ -1803,6 +1803,9 @@ int f() {
     run_process([PYTHON, EMCC, 'main.cpp'])
     self.assertContained('1234, 1234, 4321\n', run_js('a.out.js'))
 
+  def test_sdl2_mixer(self):
+    Building.emcc(path_from_root('tests', 'sdl2_mixer.c'), ['-s', 'USE_SDL_MIXER=2'], output_filename='a.out.js')
+
   def test_libpng(self):
     shutil.copyfile(path_from_root('tests', 'pngtest.png'), 'pngtest.png')
     Building.emcc(path_from_root('tests', 'pngtest.c'), ['--embed-file', 'pngtest.png', '-s', 'USE_ZLIB=1', '-s', 'USE_LIBPNG=1'], output_filename='a.out.js')

--- a/tools/ports/sdl2_mixer.py
+++ b/tools/ports/sdl2_mixer.py
@@ -37,11 +37,7 @@ def get(ports, settings, shared):
                      ['external', 'native_midi', 'timidity'])
 
     # copy header to a location so it can be used as 'SDL2/'
-    dest_include_path = os.path.join(dest_path, 'include', 'SDL2')
-    os.makedirs(dest_include_path)
-    shutil.copyfile(os.path.join(source_path, 'SDL_mixer.h'),
-                    os.path.join(dest_include_path, 'SDL_mixer.h'))
-
+    ports.install_headers(source_path, pattern='SDL_*.h', target='SDL2')
     return final
 
   return [shared.Cache.get(libname, create, what='port')]

--- a/tools/system_libs.py
+++ b/tools/system_libs.py
@@ -1553,8 +1553,10 @@ class Ports(object):
     if target:
       dest = os.path.join(dest, target)
       shared.safe_ensure_dirs(dest)
-    for f in glob.glob(os.path.join(src_dir, pattern)):
-      logger.debug(os.path.join(dest, os.path.basename(f)))
+    matches = glob.glob(os.path.join(src_dir, pattern))
+    assert matches, "no headers found to install in %s" % src_dir
+    for f in matches:
+      logger.debug('installing: ' + os.path.join(dest, os.path.basename(f)))
       shutil.copyfile(f, os.path.join(dest, os.path.basename(f)))
 
   @staticmethod

